sdk/identity/azure-identity/src/main/java/com/azure/identity/VisualStudioCodeCredential.java

```diff
@@ -112,4 +119,27 @@ public Mono<AccessToken> getToken(TokenRequestContext request) {
                 LoggingUtil.logTokenError(LOGGER, identityClient.getIdentityClientOptions(), request, other);
             });
     }
+
+    TokenCredential getBrokerAuthCredential(String clientId) {
+        if (!isVsCodeBrokerAuthAvailable()) {
+            return null;
+        }
+        try {
+            Class<?> builderClass = Class.forName(BROKER_BUILDER_CLASS);
+            Object builder = builderClass.getConstructor().newInstance();
```

Comment:
If this area ever becomes a hot path we'd want to look at leveraging MethodHandles, which we cache, over classic reflection as that can circumvent a lot of SecurityManager checking in older versions of Java